storage/remote/queue_manager.go

```diff
@@ -1669,13 +1669,12 @@ func sendWriteRequestWithBackoff(ctx context.Context, cfg config.QueueConfig, l
 	}
 }
 
-func buildWriteRequest(timeSeries []prompb.TimeSeries, metadata []prompb.MetricMetadata, pBuf *proto.Buffer, buf []byte, filter func(time.Duration, prompb.TimeSeries) bool, ageLimit time.Duration) ([]byte, int64, error) {
+func buildTimeSeries(timeSeries []prompb.TimeSeries, filter func(time.Duration, prompb.TimeSeries) bool, ageLimit time.Duration) (int64, []prompb.TimeSeries) {
 	var highest int64
-	var removeIdx []int
 
+	writeIdx := 0
```

Comment:
can we use a more descriptive name? `removeIdx` was good before, maybe `keepIdx` now?